meta_package_manager/managers/composer.py

```diff
@@ -0,0 +1,196 @@
+
+        .. code-block:: shell-session
+
+            $ composer global show --format=json | jq
+            {
+              "installed": [
+                {
+                  "name": "carbondate/carbon",
+                  "version": "1.33.0",
+                  "description": "A simple API extension for DateTime."
+                },
+                {
+                  "name": "guzzlehttp/guzzle",
+                  "version": "6.3.3",
+                  "description": "Guzzle is a PHP HTTP client library"
+                },
+                {
+                  "name": "guzzlehttp/promises",
+                  "version": "v1.3.1",
+                  "description": "Guzzle promises library"
+                },
+                {
+                  "name": "guzzlehttp/psr7",
+                  "version": "1.4.2",
+                  "description": "PSR-7 message (...) methods"
+                },
+            (...)
+        """
+        installed = {}
+
+        output = self.run([self.cli_path] + self.cli_args + [
+            'show', '--format=json'])
+
+        if output:
+
+            package_list = json.loads(output)
+            for package in package_list['installed']:
+                package_id = package['name']
+                installed[package_id] = {
+                    'id': package_id,
+                    'name': package_id,
+                    'installed_version': package['version']}
+
+        return installed
+
+    def search(self, query):
+        """ Fetch matching packages from ``composer search`` output.
+
+        Raw CLI output samples:
+
+        .. code-block:: shell-session
+
+            $ composer search symfony
+            symfony/symfony The Symfony PHP framework
+            symfony/yaml Symfony Yaml Component
+            symfony/var-dumper Symfony (...) dumping PHP variables
+            symfony/translation Symfony Translation Component
+            symfony/routing Symfony Routing Component
+            symfony/process Symfony Process Component
+            symfony/polyfill-php70 Symfony (...) features to lower PHP versions
+            symfony/polyfill-mbstring Symfony (...) Mbstring extension
+            symfony/polyfill-ctype Symfony polyfill for ctype functions
+            symfony/http-kernel Symfony HttpKernel Component
+            symfony/http-foundation Symfony HttpFoundation Component
+            symfony/finder Symfony Finder Component
+            symfony/event-dispatcher Symfony EventDispatcher Component
+            symfony/debug Symfony Debug Component
+            symfony/css-selector Symfony CssSelector Component
+        """
+        matches = {}
+
+        output = self.run([self.cli_path] + self.cli_args + [
+            'search', query])
+
+        if output:
+            regexp = re.compile(r'(\S+\/\S+) .*')
+            for package in output.split('\n'):
+                match = regexp.match(package)
+                if match:
+                    package_id = match[1]
```

Comment:
You should fetch the group here:
```suggestion
                    package_id = match.group(1)
```